### PR TITLE
B15 Heavy Assault Armor

### PR DESCRIPTION
**Pull request by Mister-moon1**
## About The Pull Request
Adds a new armor style to req for 4000 dollars a set and to comtech/RTO vendors for 30 points and 25 points respectively 
The actual armor is a muffled down version of B18 (calm your fears), internal damage, bomb and bullet protection that the B18 grants, instead giving more bio protection while still retaining quite the protection. 
This does come at the downside of speed, actually being slower than the B18.

The idea and original sprites are credited to cnaper dodo#8036 

## Why It's Good For The Game
Engis have alot to do for fortification upkeep but just get attacked at every moment as they do so, this will add a way for them to perform upkeep a bit easier.
RTO spend quite alot of time standing still for lazing, calling people and step in for comtechs when needed as they have the skills
Now the armor doesn't make you immortal by any means, you are still very much killable, its just a tier above other options defense wise.

## Changelog

:cl:
add: B15 armor and helmet
add: Added the B15 Heavy assault armor to req ASRS, RTO and comtech vendors
/:cl:

### CODE REVIEW

**Comment by AnCopper** [comment]
This is a good idea but I don't see this being used by ComTechs sine they'll be spending their points on literally everything else.

**Comment by Mister-moon1** [comment]
The helmet now comes with a toggleable visor, when the visor is up it retains the stats of a default m10 helmet, when it is down it gains more protection at the cost of vision, it does also grant minor eye protection
it has lost its hug protection since the code just allows you to flip the visor up and down to reset the hugged variable
the asrs cost and engi squad prep has also been lowered to make it more accessible since it wouldnt see much use otherwise

File: code/datums/supply_packs/armor.dm [hunk 1]
```diff
@@ -75,3 +75,12 @@
 	containername = "M4 pattern marine armor crate"
 	group = "Armor"
 
+/datum/supply_packs/armor_heavyassault
+	name = "B15 heavy assault armor crate (x1 helmet, x1 armor)"
+	contains = list(/obj/item/clothing/suit/storage/marine/heavyassault,
+					/obj/item/clothing/head/helmet/marine/heavyassault)
```
Comment (Geevies) [inline code/datums/supply_packs/armor.dm:81]:
```suggestion
	contains = list(/obj/item/clothing/suit/storage/marine/heavy_assault, /obj/item/clothing/head/helmet/marine/heavy_assault)
```

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
 	desc = "An experimental brain-bucket. A dust ruffle hangs from back. Moderately better at deflecting blunt objects at the cost of humiliation. But who will be laughing at the memorial? Not you, you'll be busy getting medals for your intel work."
 	specialty = "XM12 pattern intel"
 
+/obj/item/clothing/head/helmet/marine/heavyassault
+	name = "\improper B15 helmet"
+	desc = "A robust hunk of metal and rubber designed to go with the B15 heavy assault armor. It has a built in visor which can slide up and down to better protect yourself."
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:596]:
```suggestion
	desc = "A robust hunk of metal and rubber designed to go with the B15 heavy assault armor. It has a built in visor which can slide up and down to better protect the wearer."
```

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+	icon_state = "heavyassault"
+	item_state = "heavyassault"
+	flags_atom = FPRINT|CONDUCT
+	flags_inventory = COVEREYES|COVERMOUTH|BLOCKSHARPOBJ
+	flags_inv_hide = HIDEEARS|HIDEEYES|HIDEFACE
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:601]: if the helmet starts open, these shouldn't be set yet

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+	siemens_coefficient = 0.9
+	force = 20
+	specialty = "B15"
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:604]:
```suggestion
	specialty = "B15 pattern assault"
```

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+	eye_protection = 1
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:605]: if the helmet starts open, this shouldn't be set yet

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+	vision_impair = VISION_IMPAIR_HIGH
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:606]: if the helmet starts open, this shouldn't be set yet

File: code/modules/clothing/head/helmet.dm [hunk 7]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+	var/protection_on = FALSE
+	var/base_icon_state
+
+	actions_types = list(/datum/action/item_action/toggle)
+	vision_impair = VISION_IMPAIR_NONE
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:611]: remove this

File: code/modules/clothing/head/helmet.dm [hunk 8]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+	set name = "Toggle visor"
+	set src in usr
+
+	if(usr.canmove && !usr.stat && !usr.is_mob_restrained())
+		if(protection_on)
+			vision_impair = VISION_IMPAIR_NONE
+			flags_inventory &= ~(COVEREYES|COVERMOUTH)
+			flags_inv_hide &= ~(HIDEEYES|HIDEFACE)
+			icon_state = base_icon_state
+			eye_protection = 0
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:632]:
```suggestion
			eye_protection = FALSE
```

File: code/modules/clothing/head/helmet.dm [hunk 9]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+			to_chat(usr, "You push the visor out of your face.")
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:633]:
```suggestion
			to_chat(usr, SPAN_NOTICE("You push the visor out of your face."))
```

File: code/modules/clothing/head/helmet.dm [hunk 10]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+			armor_melee = CLOTHING_ARMOR_MEDIUM
+			armor_bullet = CLOTHING_ARMOR_MEDIUM
+			armor_laser = CLOTHING_ARMOR_MEDIUMLOW
+			armor_energy = CLOTHING_ARMOR_NONE
+			armor_bomb = CLOTHING_ARMOR_LOW
+			armor_bio = CLOTHING_ARMOR_MEDIUM
+			armor_rad = CLOTHING_ARMOR_LOW
+			armor_internaldamage = CLOTHING_ARMOR_MEDIUM
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:641]: reset to initial rather than hardcoding it here

File: code/modules/clothing/head/helmet.dm [hunk 11]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+		else
+			vision_impair = VISION_IMPAIR_HIGH
+			flags_inventory |= COVEREYES|COVERMOUTH
+			flags_inv_hide |= HIDEEYES|HIDEFACE
+			icon_state = "[base_icon_state]_down"
+			eye_protection = 1
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:647]:
```suggestion
			eye_protection = TRUE
```

File: code/modules/clothing/head/helmet.dm [hunk 12]
```diff
@@ -591,6 +591,83 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+			to_chat(usr, "You slide the visor down to protect your face.")
```
Comment (Geevies) [inline code/modules/clothing/head/helmet.dm:648]:
```suggestion
			to_chat(usr, SPAN_NOTICE("You slide the visor down to protect your face."))
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 13]
```diff
@@ -564,6 +564,22 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	slowdown = SLOWDOWN_ARMOR_LOWHEAVY
 	movement_compensation = SLOWDOWN_ARMOR_MEDIUM
 
+/obj/item/clothing/suit/storage/marine/heavyassault
```
Comment (Geevies) [inline code/modules/clothing/suits/marine_armor.dm:567]:
```suggestion
/obj/item/clothing/suit/storage/marine/heavy_assault
```

File: code/modules/cm_marines/equipment/kit_boxes.dm [hunk 14]
```diff
@@ -520,3 +520,11 @@
 	new /obj/item/storage/backpack/marine/smock(src)
 	new /obj/item/device/binoculars/range/designator/spotter(src)
 	new /obj/item/pamphlet/skill/spotter(src)
+
+/obj/item/storage/box/kit/heavyassault
+	name = "\improper B15 Heavy Assault Armor kit"
+	pro_case_overlay = "engi"
+
+/obj/item/storage/box/kit/heavyassault/fill_preset_inventory()
+	new /obj/item/clothing/suit/storage/marine/heavyassault(src)
```
Comment (Geevies) [inline code/modules/cm_marines/equipment/kit_boxes.dm:529]:
```suggestion
	new /obj/item/clothing/suit/storage/marine/heavy_assault(src)
```

File: code/modules/cm_marines/equipment/kit_boxes.dm [hunk 15]
```diff
@@ -520,3 +520,11 @@
...
+	new /obj/item/clothing/head/helmet/marine/heavyassault(src)
```
Comment (Geevies) [inline code/modules/cm_marines/equipment/kit_boxes.dm:530]:
```suggestion
	new /obj/item/clothing/head/helmet/marine/heavy_assault(src)
```

File: code/modules/cm_marines/equipment/kit_boxes.dm [hunk 16]
```diff
@@ -520,3 +520,11 @@
 	new /obj/item/storage/backpack/marine/smock(src)
 	new /obj/item/device/binoculars/range/designator/spotter(src)
 	new /obj/item/pamphlet/skill/spotter(src)
+
+/obj/item/storage/box/kit/heavyassault
```
Comment (Geevies) [inline code/modules/cm_marines/equipment/kit_boxes.dm:524]:
```suggestion
/obj/item/storage/box/kit/heavy_assault
```

File: code/modules/cm_marines/equipment/kit_boxes.dm [hunk 17]
```diff
@@ -520,3 +520,11 @@
...
+	name = "\improper B15 Heavy Assault Armor kit"
+	pro_case_overlay = "engi"
+
+/obj/item/storage/box/kit/heavyassault/fill_preset_inventory()
```
Comment (Geevies) [inline code/modules/cm_marines/equipment/kit_boxes.dm:528]:
```suggestion
/obj/item/storage/box/kit/heavy_assault/fill_preset_inventory()
```

File: code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm [hunk 18]
```diff
@@ -72,6 +72,7 @@ GLOBAL_LIST_INIT(cm_vending_gear_engi, list(
 		list("JTAC Pamphlet", 15, /obj/item/pamphlet/skill/jtac, null, VENDOR_ITEM_REGULAR),
 		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
 		list("Large Shotgun Shell Pouch", 10, /obj/item/storage/pouch/shotgun/large, null, VENDOR_ITEM_RECOMMENDED),
+		list("B15 Heavy Assault Armor Kit", 30, /obj/item/storage/box/kit/heavyassault, null, VENDOR_ITEM_REGULAR),
```
Comment (Geevies) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm:75]:
```suggestion
		list("B15 Heavy Assault Armor Kit", 30, /obj/item/storage/box/kit/heavy_assault, null, VENDOR_ITEM_REGULAR),
```

**Review by Geevies — CHANGES_REQUESTED** [review]
make sure these armors either don't use camo variants by setting NO_SNOW_VARIANT, or sprite camo variants for them

**Review by Geevies — APPROVED** [review]
code approved, we'll see about balance/lore

File: code/modules/clothing/head/helmet.dm [hunk 19]
```diff
@@ -591,6 +591,74 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
...
+
+/obj/item/clothing/head/helmet/marine/heavy_assault/Initialize()
+	. = ..()
+	base_icon_state = icon_state
+
+/obj/item/clothing/head/helmet/marine/heavy_assault/attack_self(mob/user)
+	..()
+	toggle()
+
+/obj/item/clothing/head/helmet/marine/heavy_assault/verb/toggle()
```
Comment (carlarctg) [inline code/modules/clothing/head/helmet.dm:617]: why is this a verb? it already has an action attached, it's redundant and we want to limit the amount of verbs we have. turn it into a normal proc and replace usr

Comment (spookydonut) [reply code/modules/clothing/head/helmet.dm:617]: Don't use a verb like this.
Use an action button with a keybinding if you want it able to be activated by a keypress

File: code/modules/clothing/head/helmet.dm [hunk 20]
```diff
@@ -614,7 +614,7 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
 	..()
 	toggle()
 
-/obj/item/clothing/head/helmet/marine/heavy_assault/verb/toggle()
+/obj/item/clothing/head/helmet/marine/heavy_assault/proc/toggle()
 	set category = "Object"
 	set name = "Toggle visor"
 	set src in usr
```
Comment (harryob) [inline code/modules/clothing/head/helmet.dm:620]: proc no need these

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Pyrodes** [comment]
Yes!!!!1 Become the tank if you cannot have it!!!!!!!

**Comment by Mister-moon1** [comment]
PR is on hold while i update the sprites, gonna take quite a few days

**Comment by Mister-moon1** [comment]
minor fuckup including removing all commits, not to worry as we are back on track hopefully without conflicts

**Comment by monkeysfist101** [comment]
The sprites are oddly asymmetric and do not look like a natural progression of our light->medium->heavy armor. The USCM equivalent of medieval full plate is kind of odd in general, too. 

Sprites aside, I spoke to the other devs and we're denying this on balance reasons. This fills a niche that didn't need to be filled.

**Pull request closed without merging**